File: AdminArea/AdminArea.Web/Modules/souqcom/Categorycom/CategorycomColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminArea.souqcom.Columns;

[ColumnsScript("souqcom.Categorycom")]
[BasedOnRow(typeof(CategorycomRow), CheckNames = true)]
public class CategorycomColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Image { get; set; }
    public string Description { get; set; }
}

File: AdminArea/AdminArea.Web/Modules/ServerTypes/souqcom/CategorycomColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CategorycomRow } from "./CategorycomRow";

export interface CategorycomColumns {
    Id: Column<CategorycomRow>;
    Name: Column<CategorycomRow>;
    Image: Column<CategorycomRow>;
    Description: Column<CategorycomRow>;
}

export class CategorycomColumns extends ColumnsBase<CategorycomRow> {
    static readonly columnsKey = 'souqcom.Categorycom';
    static readonly Fields = fieldsProxy<CategorycomColumns>();
}